### PR TITLE
Backend: Refactor beckn latency PR

### PR DESCRIPTION
**Pull request by DonDokidon**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

A bit less redundancy, overall easier to work with

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Cancel.hs [hunk 1]
```diff
@@ -45,17 +45,14 @@ cancel ::
...
-    dCancelReq <- ACL.buildCancelReq req
-    case dCancelReq of
-      Left cancelReq -> do
-        Redis.whenWithLockRedis (cancelLockKey cancelReq.bookingId.getId) 60 $ do
-          (merchant, booking) <- DCancel.validateCancelRequest transporterId subscriber cancelReq
-          fork ("cancelBooking:" <> cancelReq.bookingId.getId) $
-            DCancel.cancel cancelReq merchant booking
-      Right cancelSearchReq -> do
-        searchReq <- DCancel.validateCancelSearchRequest transporterId subscriber cancelSearchReq
-        fork ("cancelSearch:" <> cancelSearchReq.transactionId) $
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Cancel.hs:57]: Why did you remove the fork here? And why to revert all these changes they were done to reduce the latency, after you refactoring it will be the same code it was before.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Cancel.hs (hunk 1 above)
Comment (DonDokidon) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Cancel.hs:57]: > Why did you remove the fork here? And why to revert all these changes they were done to reduce the latency, after you refactoring it will be the same code it was before.

I did not remove it, [i moved it in domain logic](https://github.com/nammayatri/nammayatri/pull/985/files/3eae37014dc8846d5372658f176d6ad6021dc87a#diff-3359acbc2958c5b0634d3cdb45da8ffbd7be4674fc0e0a2c36bc08db9c0fb24fR88). It's much easier to work with that way, no redundant data types.
